opentelemetry_collector/receiver/nginxreceiver/factory.go

```diff
@@ -0,0 +1,68 @@
+package nginxreceiver
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configerror"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer"
+)
+
+const (
+	typeStr = "nginxstats"
+)
+
+// Factory is the factory for the nginx stats receiver.
+type Factory struct {
+}
+
+// Type gets the type of the Receiver config created by this factory.
+func (f *Factory) Type() configmodels.Type {
+	return typeStr
+}
+
+// CustomUnmarshaler returns a custom unmarshaler for this config.
+// Returning nil means that this receiver does not use one.
+func (f *Factory) CustomUnmarshaler() component.CustomUnmarshaler {
+	return nil
+}
+
+// CreateDefaultConfig creates the default configuration for the receiver.
+func (f *Factory) CreateDefaultConfig() configmodels.Receiver {
```

Comment:
Is there an expectation that DefaultConfig should be usable? Without statsURL defined, that doesn't seem to be possible. So this config would be useless?

Reply:
The DefaultConfig is required by the ReceiverFactory interface. I think it's also used to set the ReceiverSettings value for receivers even when they do have an explicit config.